packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -12,11 +13,27 @@ import 'package:flutter/widgets.dart';
 import 'colors.dart';
 import 'theme.dart';
 import 'typography.dart';
+import 'constants.dart';
 
+const Duration _kDialAnimateDuration = const Duration(milliseconds: 200);
+const double _kTwoPi = 2 * math.PI;
+const int _kHoursPerDay = 24;
+const int _kHoursPerPeriod = 12;
+const int _kMinutesPerHour = 60;
+
+enum DayPeriod {
+  am,
+  pm,
+}
+
+/// A value representing a time during the day
 class TimeOfDay {
   const TimeOfDay({ this.hour, this.minute });
 
+  /// Returns a new TimeOfDay with the hour and/or minute replaced.
   TimeOfDay replacing({ int hour, int minute }) {
+    assert(hour == null || (hour >= 0 && hour < _kHoursPerDay));
```

Comment:
you assert this here, but not in the constructor, which is amusing. Not sure how to fix that without compile-time constant asserts though. :-(